Merge Tempestar11:main fork to main

Tempestar11 opened the pull request:
- Added NPC appraisal lookup
- Added NPC 90% appraisal lookup
- Added Jita 90% appraisal lookup

Tempestar11 committed:
Added jita 90% field to discord embed

Tempestar11 committed:
Revert "Added jita 90% field to discord embed"

This reverts commit b8b65f0fc54ce6b211b6453435a68436ec7cadf9.

Tempestar11 committed:
Added 90% Jita buy to discord embed

Tempestar11 committed:
Same as last commit

Tempestar11 committed:
Added an API request for 90% Jita buy + differentiated different requests

Tempestar11 committed:
Added additional 90% Jita buy API request to discord apprasial and recall embed

Tempestar11 committed:
merged the API requests into one function with an additional percentage field

Tempestar11 committed:
Added docstring to API request function

Tempestar11 committed:
Added NPC market to API request and discord embed

Tempestar11 committed:
Merge pull request #2 from Tempestar11/jita-buy-90%

Jita buy 90%

Tempestar11 committed:
Merge branch 'main' into add-NPC-station-lookup

Tempestar11 committed:
Merge pull request #1 from Tempestar11/add-NPC-station-lookup

Added NPC market to API request and discord embed

Tempestar11 committed:
fixed arguments for API requests

Tempestar11 committed:
added NPC 90% buy embed

Tempestar11 committed:
Rearranged NPC lookups to solve partial string matching

Tempestar11 committed:
Merge branch 'main' into npc-buy-90%

Tempestar11 committed:
Merge pull request #3 from Tempestar11/npc-buy-90%

Npc buy 90%

Tempestar11 committed:
Normalised regex

m0dman commented on Program.cs:
```diff
@@ -63,9 +64,10 @@ private async Task MessageReceived(SocketMessage message)
             if (message.Author.IsBot) return;
 
-            if (message.Content.StartsWith("!appraise"))
+            if (Regex.IsMatch(message.Content, @"^!appraise\b"))
```
Instead of regex could you not use .Equals or ==? As .StartsWith would be causing the partial string matching

Tempestar11 replied:
I'm not too sure what you mean by that. What I think you're saying is to take a substring from the message content and match it that way?
I also figured using regex this way might open up ways for custom percentages by adding whatever number behind the command.

m0dman replied:
You can do if (message.Content.Equals("!appraise")) or  if (message.Content == "!appraise") and this should resolve the partial string matching issue without needing regex 
But actually I do like that idea grabbing the number behind the command to then do custom percentages

m0dman commented on Program.cs:
```diff
@@ -165,20 +175,141 @@ private async Task MessageReceived(SocketMessage message)
         }
 
-        private async Task<JsonElement> GetAppraisal(string items)
+        private async Task<JsonElement> GetAppraisal(string items, float percentage, int market)
```
I understand that as it stands this is functioning as intended with a float percentage but I believe the Janice API is meant to accept a double value for the pricePercentage instead of float

m0dman reviewed (changes requested):
All good just a couple of suggestions :)

Tempestar11 committed:
Updated data type for API request

Updated data type for price percentage arguments passed to the API request function to double instead of floats

Tempestar11 committed:
Merge pull request #4 from Tempestar11/Tempestar11-patch-1

Updated data type for API request

m0dman commented on Program.cs:
```diff
@@ -200,7 +200,7 @@ private async Task MessageReceived(SocketMessage message)
                     
 
-                    var appraisal = await GetAppraisal(requestBody, 0.9f, 6);
+                    var appraisal = await GetAppraisal(requestBody, 0.9d, 6);
```
should this not just be 90 now instead of 0.9d since its now double?

Tempestar11 committed:
Updated double values being passed to Janice API

Tempestar11 committed:
Merge pull request #5 from Tempestar11/Tempestar11-patch-2

Updated double values being passed to Janice API
